### PR TITLE
Documentation setup (fix #121, #188 and #198)

### LINKED ISSUE #121: Do doc setup
- [ ] see how the documentation files are managed in DNF https://github.com/rpm-software-management/dnf/tree/master/doc
- [ ] set the CMakefile to generate hmtl and man pages
- [ ] include man pages in spec
- [ ] add Sphinx requirement in spec

### PR DESCRIPTION
**Pull request by LukasSlouka**

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+# If false, no module index is generated.
+#latex_domain_indices = True
+
+
+# -- Options for manual page output ---------------------------------------
+
+# One entry per manual page. List of tuples
+# (source start file, name, description, authors, manual section).
+man_pages = [
+    #('index', 'rpg', u'rpg Documentation',
```
Comment (jsilhan) [inline doc/conf.py]: delete the commented lines, please

File: doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Welcome to rpg's documentation!
+===============================
+
+Contents:
+
+.. toctree::
+   :maxdepth: 1
+
+   command_ref
+   conf_ref
```
Comment (jsilhan) [inline doc/index.rst]: conf_ref is not there, right?

**Comment by LukasSlouka** [comment]
yeah i forgot to remove this one, sorry

**Comment by LukasSlouka** [comment]
Done

**Comment by LukasSlouka** [comment]
I hope I did not forget anything, however I did not put it in separate section (eg. API like it is in dnf). Anyway it shows up nicely in generated html documentation.

**Comment by LukasSlouka** [comment]
Sorry for the delay, it is autodocumenting now. However, it required some changes in spec.py file, because it was throwing syntax errors that prevented documentation build.

**Comment by LukasSlouka** [comment]
I'll put changes in spec.py in separate commit

**Comment by LukasSlouka** [comment]
Done :)

**Comment by jsilhan** [comment]
Do `git rebase origin/master` and fix the conflicts, please.

> I'll put changes in spec.py in separate commit

That was good choice

**Comment by LukasSlouka** [comment]
Apparently it needs python-sphinx and does not need python3-sphinx. I am little bit confused, because sphinx manuals does not mention any particular differences in usage of these 2 packages, yet python3-sphinx package does not provide sphinx-build command by itself. I will try to look deeper into it

**Comment by LukasSlouka** [comment]
I take that back, now it is completely independent from `python-sphinx` and should work just fine

**Comment by jsilhan** [comment]
still getting merge conflict error. Ask @cibo94 for help if you don't know how to rebase to upstream.

**Comment by LukasSlouka** [comment]
I had `This branch is up-to-date with the base branch` message before, don't know what changed. Should be okay now

**Comment by jsilhan** [comment]
Perfect, merged, thanks.

**Pull request merged**